OpenRA.Mods.Common/Widgets/ObserverProductionIconsWidget.cs

```diff
@@ -132,5 +162,31 @@ public override Widget Clone()
 		{
 			return new ObserverProductionIconsWidget(this);
 		}
+
+		public override void MouseEntered()
+		{
+			if (TooltipContainer != null)
+				tooltipContainer.Value.SetTooltip(TooltipTemplate,
+					new WidgetArgs() { { "world", world }, { "player", GetPlayer() }, { "getTooltipIcon", GetTooltipIcon } });
+		}
+
+		public override void MouseExited()
+		{
+			if (TooltipContainer == null)
+				return;
+
+			tooltipContainer.Value.RemoveTooltip();
+		}
+
+		public override bool HandleMouseInput(MouseInput mi)
```

Comment:
This should change to `Tick`, otherwise the tooltip will not update as the icons change under the mouse.  You can replace `mi.Location` &rarr; `Viewport.LastMousePos`.